crates/rpc-types-eth/src/state.rs

```diff
@@ -30,6 +31,64 @@ pub struct AccountOverride {
     pub state_diff: Option<HashMap<B256, B256>>,
 }
 
+/// Helper type that bundles various overrides for EVM Execution.
+///
+/// By `Default`, no overrides are included.
+#[derive(Debug, Clone, Default)]
+pub struct EvmOverrides {
+    /// Applies overrides to the state before execution.
+    pub state: Option<StateOverride>,
+    /// Applies overrides to the block before execution.
+    ///
+    /// This is a `Box` because less common and only available in debug trace endpoints.
+    pub block: Option<Box<BlockOverrides>>,
+}
+
+impl EvmOverrides {
+    /// Creates a new instance with the given overrides
+    pub const fn new(state: Option<StateOverride>, block: Option<Box<BlockOverrides>>) -> Self {
+        Self { state, block }
+    }
+
+    /// Creates a new instance with the given state overrides.
+    pub const fn state(state: Option<StateOverride>) -> Self {
```

Comment:
why even allow `None` as an input?

Reply:
this is also mainly useful from server POV because some methods only support `StateOverrides` or `BlockOverrides` and usually as Option.

Reply:
My main point is that passing `None` to this function is always more clearly represented as `Default::default` which means this API is an artifact of a specific usage, not a good API on its own, and is not something we should make available to non-reth users

```
// `fn state(state:StateOverride) -> Self`
state_opt.map(EvmOverrides::state).unwrap_or_default()

// current api
EvmOverrides::state(state_opt)

// From<Option<_>> feels cleanest
state_opt.into()
```

Reply:
```
/// Matches some other spots in reth
EvmOverrides::default().maybe_state(state_opt)
```